provider/core/src/response.rs

```diff
@@ -127,36 +127,36 @@ where
 
 /// A wrapper type that wraps either an [`Rc`](alloc::rc::Rc) or an
 /// [`Arc`](alloc::sync::Arc), depending on the "sync" feature. Create
-/// this from a `&[u8]`.
+/// this from a `&[T]`.
 #[derive(Clone)]
-pub struct RcWrap(
-    #[cfg(not(feature = "sync"))] alloc::rc::Rc<[u8]>,
-    #[cfg(feature = "sync")] alloc::sync::Arc<[u8]>,
+pub struct RcWrap<T>(
+    #[cfg(not(feature = "sync"))] alloc::rc::Rc<[T]>,
+    #[cfg(feature = "sync")] alloc::sync::Arc<[T]>,
 );
 
-impl core::ops::Deref for RcWrap {
-    type Target = [u8];
+impl<T> core::ops::Deref for RcWrap<T> {
+    type Target = [T];
```

Comment:
You should use `T` instead of `[T]`, your current code works for arbitrary slices but not for arbitrary types.

Reply:
Yes. To clarify, call sites should say `RcWrap<[u8]>` rather than `RcWrap<u8>`

Reply:
Yeah. I actually started with what you suggested. For that to work, the trait `Clone` has to be implemented for type T. Note that `#[derive(Clone)]` is specified for RcWrap struct because it is required by the bound in `yoke::CloneableCart`:
`pub unsafe trait CloneableCart: Clone {}`.

For RcWrap<[u8], Clone has to be implemented for type [u8], but that is not provided because [u8] doesn't have a fixed size (https://stackoverflow.com/questions/68215776/why-u8-does-not-implement-clone). I guess I would need to provide a custom Clone implementation for RcWrap<[u8]> to get around it? I tried a few things, but still couldn't wrap my head around this.